File: GO/88.go
func merge(nums1 []int, m int, nums2 []int, n int) {
	ans := make([]int, 0, m+n)
	i1, i2 := 0, 0
	for i1 < m && i2 < n {
		if nums1[i1] < nums2[i2] {
			ans = append(ans, nums1[i1])
			i1++
		} else {
			ans = append(ans, nums2[i2])
			i2++
		}
	}
	for i1 < m {
		ans = append(ans, nums1[i1])
		i1++
	}
	for i2 < n {
		ans = append(ans, nums2[i2])
		i2++
	}
	for i, v := range ans {
		nums1[i] = v
	}
}

File: JAVA/88.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int[] ans = new int[m+n];
        int i1 = 0,i2 = 0, cur = 0;
        while(cur < m+n && i1 < m && i2 < n) {
            if(nums1[i1] < nums2[i2]) ans[cur++] = nums1[i1++];
            else ans[cur++] = nums2[i2++];
        }
        while(i1 < m) ans[cur++] = nums1[i1++];
        while(i2 < n) ans[cur++] = nums2[i2++];
        for(int i = 0;i < m+n;i ++) nums1[i] = ans[i];
    }
}